File: DS/CPP/CallByValue.cpp
#include<iostream>
void swap(int x, int y);		//protype is very important

using namespace std;

int main()
{
    int a=20, b=10;
    swap(a, b);
    cout<<"Values of a and b after swapping "<<a<<", "<<b<<", respectively."<<endl;
}

void swap(int x, int y)
{
    int t = x;
    x = y;
    y = t;
    //cout<<"Values of a and b after swapping "<<x<<", "<<y<<", respectively."<<endl;
}

/*

	Swapping didn't occur as the the values of a and b we copied
	to x and y.

*/

File: DS/Java/CallByValue.java
public class CallByValue {
	public static void main(String[] args) {
		int a = 20, b = 10;
		swap(a, b);		//Error about static content
		System.out.println("The values of a and b after swapping are " + a + ", " + b + ", respectively.");
	}
	static void swap(int x, int y) {
		int t = x;
		x = y;
		y = x;	
	}
}
